src/Cli/dotnet/commands/dotnet-workload/update/WorkloadUpdateCommand.cs

```diff
@@ -180,10 +229,13 @@ public void UpdateWorkloads(bool includePreviews = false, DirectoryPath? offline
                     {
                         _workloadInstaller.RemoveManifestsFromInstallState(sdkFeatureBand);
                     }
+
+                    _workloadInstaller.AdjustWorkloadSetInInstallState(sdkFeatureBand, workloadSetVersion);
```

Comment:
It looks like whenever we update to a new workload set, we write the workload set version to the install state (I think this applies to both the update and install commands).

We actually don't normally want to write the workload set version to the install state.  The reason to have a workload set version in the install state is to pin to that version, which should only happen if the `--version` parameter is explicitly specified.  Once the version is pinned, a workload install command should by default not update to a newer version.  A `dotnet workload update --version <version>` command should update the pinned version in the install state, and `dotnet workload update` should remove the pinned version and update to the latest workload set version available.

Reply:
We can talk about this one. I find it very valuable to have the workload set version in the install state file 100% of the time after updating or installing via a workload set. It doesn't have the same 'pinned' function as it does for workload manifests because whereas you can update some workload manifests without updating the pinned workload manifests, the same isn't true for workload sets: there's only one workload set. If you do a `dotnet workload update`, it updates the one and only workload set. If you do `dotnet workload update --version <version>`, it updates the workload set. There isn't really the same kind of scenario where you wouldn't want to update some workload set whereas you would want to update another one if that makes sense.